catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/forms/SearchFormsLoader.java

```diff
@@ -100,33 +103,43 @@ public static boolean enabled() {
    * <p>Caution should be exercised when executing code as system. Results from querying as system
    * are never returned to the client or cached. This means there is no risk of data leak.
    *
-   * @param framework the catalog framework, for creating system templates.
-   * @param util for querying the catalog.
+   * @param catalogFramework the catalog framework, for creating system templates.
+   * @param endpointUtil for querying the catalog.
    * @param systemTemplates system templates loaded from config.
    */
   public static void bootstrap(
-      CatalogFramework framework, EndpointUtil util, List<Metacard> systemTemplates) {
-    Function<Map<String, Result>, Set<String>> transform = SearchFormsLoader::titlesTransform;
-    Set<String> queryTitles =
-        executeAsSystem(() -> transform.apply(util.getMetacardsByFilter(QUERY_TEMPLATE_TAG)));
-    Set<String> resultTitles =
-        executeAsSystem(() -> transform.apply(util.getMetacardsByFilter(ATTRIBUTE_GROUP_TAG)));
-
-    List<Metacard> dedupedTemplateMetacards =
-        Stream.concat(
-                systemTemplates
-                    .stream()
-                    .filter(QueryTemplateMetacard::isQueryTemplateMetacard)
-                    .filter(metacard -> !queryTitles.contains(metacard.getTitle())),
-                systemTemplates
-                    .stream()
-                    .filter(AttributeGroupMetacard::isAttributeGroupMetacard)
-                    .filter(metacard -> !resultTitles.contains(metacard.getTitle())))
-            .collect(Collectors.toList());
-
-    if (!dedupedTemplateMetacards.isEmpty()) {
-      executeAsSystem(() -> framework.create(new CreateRequestImpl(dedupedTemplateMetacards)));
+      CatalogFramework catalogFramework,
+      EndpointUtil endpointUtil,
+      List<Metacard> systemTemplates) {
+    Subject systemSubject = SECURITY.runAsAdmin(SECURITY::getSystemSubject);
```

Comment:
will the credentials need to be refreshed?

Reply:
I doubt it; this loading only occurs at installation once and never again